gateway-js/src/supergraphManagers/UplinkSupergraphManager/index.ts

```diff
@@ -179,9 +179,7 @@ export class UplinkSupergraphManager implements SupergraphManager {
       this.logger.debug(
         'Error fetching supergraphSdl from Uplink, calling updateSupergraphSdlFailureCallback',
       );
-      supergraphSdl = await this.onFailureToFetchSupergraphSdl.call(this, {
-        error: e,
-      });
+      supergraphSdl = await this.onFailureToFetchSupergraphSdl({ error: e });
```

Comment:
I can't leave the comment where I wanted to, but I expected a `.bind(this)` call on L96 now:

```ts
this.onFailureToFetchSupergraphSdl = onFailureToFetchSupergraphSdl.bind(this);
```

Since this is a user-provided callback, by default the `this` within their function would depend on the context (in the general sense) of where and how they wrote the function.

The typing you've provided:
```ts
export type FailureToFetchSupergraphSdlFunction = (
  this: UplinkSupergraphManager,
  // ...
)
```
...indicates to me that you want them to have access to _the `UplinkSupergraphManager` instance_ which seems reasonable to me, though I'd be interested to know more about the expected use case is these. It might not be _that_ helpful (but I'd be happy to be wrong).

I digress. Let's assume they need a handle on that instance. We've established that the default `this` within the callback currently depends on the context of where that callback was written. But really we want it to be the class instance. This is where `.bind()` becomes useful. We can decide what `this` is from inside of the function that the _user_ handed off to us. Whatever we provide as the first argument to bind is what `this` is from inside the function.

That leaves one open question: how do we tell the function what it's _overridden_ `this` type actually is? Exactly what you have above - we provide a `this: UplinkSupergraphManager` as the "first" argument to a function. At runtime it's actually _not_ the first argument, so this is a special way to indicate what the expected `this` is for a function type.

If you already knew all this stuff, well, don't mind me. If you're unsure of anything I'm happy to discuss!

Reply:
Ha, I thought your previous comment was saying the `.call` was unnecessary since passing the callback is all done within the context of `new UplinkSupergraphManager` constructor and would already know the right `this` (which will probably _usually_ be the case, but will depend on how users construct it).

Were you more saying we should call `.bind` once in the constructor instead of `.call` each time it's called?

Reply:
I had you all wrong (my recollection of `.call()` was incorrect!). What you had to begin with was fine after all, and what I'm suggesting turns out to be just a minor nitpick that accomplishes the same. Sorry for the detour.

Reply:
All good! As we discussed IRL, doing the 1x `.bind` at construction time is probably better than `.call` each time. Updated in the latest.

Reply:
Gonna stick with the "pure function" approach. Happy to revisit if that seems like a path of sadness.